stl/inc/format

```diff
@@ -1248,15 +1243,425 @@ _OutputIt _Write(_OutputIt _Out, basic_string_view<_CharT> _Value) {
+    auto _Format   = chars_format::general;
+    auto _Exponent = '\0';
+
+    switch (_Specs._Type) {
+    case 0:
+        break;
+
+    case 'A':
+        _To_upper = true;
+    case 'a':
+        _Format   = chars_format::hex;
+        _Exponent = 'p';
+        break;
+
+    case 'E':
+        _To_upper = true;
+    case 'e':
+        if (_Specs._Precision == -1)
+            _Specs._Precision = 6;
+        _Format   = chars_format::scientific;
+        _Exponent = 'e';
+        break;
+
+    case 'F':
+    case 'f':
+        if (_Specs._Precision == -1)
+            _Specs._Precision = 6;
+        _Format = chars_format::fixed;
+        break;
+
+    case 'G':
+        _To_upper = true;
+    case 'g':
+        if (_Specs._Precision == -1)
+            _Specs._Precision = 6;
+        _Format   = chars_format::general;
+        _Exponent = 'e';
+        break;
+
+    default:
+        throw format_error("invalid floating point type");
+    }
+
+    array<char, _Format_min_buffer_length> _Buffer;
+    to_chars_result _Result;
+
+    if (_Specs._Precision == -1) {
+        _Result = to_chars(_Buffer.data(), _Buffer.data() + _Buffer.size(), _Value, _Format);
+    } else {
+        _Result = to_chars(_Buffer.data(), _Buffer.data() + _Buffer.size(), _Value, _Format, _Specs._Precision);
+    }
+
+    _STL_ASSERT(_Result.ec == errc{}, "to_chars failed");
+
+    auto _Buffer_start = _Buffer.data();
+    auto _Width        = static_cast<int>(_Result.ptr - _Buffer_start);
+
+    auto _Is_negative = signbit(_Value);
+
+    if (_Is_negative) {
+        // Remove the '-', it will be dealt with directly
+        _Buffer_start += 1;
+    } else {
+        if (_Specs._Sgn != _Sign::_Minus) {
+            _Width += 1;
+        }
+    }
+
+    if (_To_upper) {
+        _Fmt_buffer_to_uppercase(_Buffer_start, _Result.ptr);
+        _Exponent = static_cast<char>(_CSTD toupper(_Exponent));
+    }
+
+    auto _Is_finite = !_STD isnan(_Value) && !_STD isinf(_Value);
+
+    auto _Append_decimal   = false;
+    auto _Exponent_start   = _Result.ptr;
+    auto _Zeroes_to_append = 0;
+
+    if (_Specs._Alt && _Is_finite) {
+        auto _Radix_point = _Result.ptr;
+        for (auto it = _Buffer_start; it < _Result.ptr; ++it) {
+            if (*it == '.') {
+                _Radix_point = it;
+            } else if (*it == _Exponent) {
+                _Exponent_start = it;
+            }
+        }
+        if (_Radix_point == _Result.ptr) {
+            ++_Width;
+            _Append_decimal = true;
+        }
+
+        if (_Specs._Type == 'g' || _Specs._Type == 'G') {
+            _Zeroes_to_append = _Specs._Precision - static_cast<int>(_Exponent_start - _Buffer_start);
+            if (!_Append_decimal)
+                _Zeroes_to_append += 1;
+            _Width += _Zeroes_to_append;
+        }
+    }
+
+    bool _Write_leading_zeroes = _Specs._Leading_zero && _Specs._Alignment == _Align::_None;
+
+    auto _Writer = [&](_OutputIt _Out) {

```

Comment:
same comment on this only being called once.

Reply:
Used twice.